migration/migrator/migrations/course/20240324130619_grade_notifications.py

```diff
@@ -0,0 +1,47 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+        """
+            ALTER TABLE notification_settings ADD COLUMN IF NOT EXISTS all_released_grades BOOLEAN DEFAULT true NOT NULL;
+            ALTER TABLE notification_settings ADD COLUMN IF NOT EXISTS all_released_grades_email BOOLEAN DEFAULT true NOT NULL;
+            ALTER TABLE gradeable ADD COLUMN IF NOT EXISTS g_notification_state BOOLEAN DEFAULT false NOT NULL;            
+            UPDATE gradeable SET g_notification_state = true WHERE g_grade_released_date < NOW(); 
+        """ 
+    )
+
+
+def down(config, database, semester, course):
+
+    """
+    Run down migration (rollback).
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+        """
+            ALTER TABLE notification_settings DROP COLUMN all_released_grades;
+            ALTER TABLE notification_settings DROP COLUMN all_released_grades_email;
+            ALTER TABLE gradeable DROP COLUMN g_notification_state;
```

Comment:
We typically don't drop columns so data isn't lost if the Submitty administrator temporarily rolls the migration back.

Reply:
Do you think I should leave this commented out or remove them?

Reply:
This should definitely be removed.  The down migration should be empty for this PR.